libraries/AP_HAL_ChibiOS/AnalogIn.cpp

```diff
@@ -366,44 +366,113 @@ void AnalogIn::update_power_flags(void)
 {
     uint16_t flags = 0;
 
-#ifdef HAL_GPIO_PIN_VDD_BRICK_VALID
-    if (!palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID)) {
+    /*
+      primary "brick" power supply valid pin. Some boards have this
+      active high, some active low. Use nVALID for active low, VALID
+      for active high
+    */
+#if defined(HAL_GPIO_PIN_VDD_BRICK_VALID)
+    if (palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID) == 1) {
+        flags |= MAV_POWER_STATUS_BRICK_VALID;
+    }
```

Comment:
This looks like something for which we can generate macro for, which converts HAL_GPIO_PIN_VDD_BRICK_VALID into (palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID) == 1) or (palReadLine(HAL_GPIO_PIN_VDD_BRICK_VALID) == 0) making it even more concise and readable.

Reply:
we could, but for now I just want to know if I have broken anything. Can you check it is correct for the Hex boards?